staging/src/k8s.io/apiserver/pkg/cel/openapi/resolver/refs.go

```diff
@@ -86,7 +86,9 @@ func populateRefs(schemaOf func(ref string) (*spec.Schema, bool), visited sets.S
 		}
 		if populated != result.AdditionalProperties.Schema {
 			changed = true
-			result.AdditionalProperties.Schema = populated
+			newProps := *result.AdditionalProperties
+			newProps.Schema = populated
+			result.AdditionalProperties = &newProps
```

Comment:
this change looks good, can you add a unit test that would have caught the issue that this mutated the original?